juju/juju
# 2.1 kvm race test fixes 1664437

This isn't critical for 2.1.0, as it is intended to be a code refactoring at testing change, but it does add tests to an area of the code that I've been working in that did not have much (if any). It also removes some of the code paths that were replaced, but left around because of what testing there *was* available.

## Description of change

PrepareHost wasn't tested directly, and even before my last refactoring, was only loosely tested. We can't test all of ContainerSetup because part of its function is to `apt install lxd` and actually deploy containers. So this is splitting off a portion of that into a helper object that can have its functionality overridden.

## QA steps

```go test```

Though if you want to make sure I haven't broken the functionality
```
  $ juju bootstrap maas
  $ juju deploy ubuntu --to lxd:0 -m controller
```

And see that the host networking is properly updated for the container.


## Documentation changes

Internal only refactoring.

## Bug reference

This was wanting to get the code tested as part of fixing [lp:1664437](https://pad.lv/1664437). It doesn't change the fix for it, but it does add tests that we are correctly interacting with a 'lock' object.
